tomviz/python/tomviz/cli/__init__.py

```diff
@@ -59,23 +65,46 @@ def main(data_file_path, state_file_path, output_file_path, progress_method,
 
     # if we have been provided a data file path we are going to use the one
     # from the state file, so check it exists.
-    if data_file_path is None:
+    if data_path is None:
         if 'reader' not in datasource:
             raise Exception('Data source does not contain a reader.')
         filenames = datasource['reader']['fileNames']
         if len(filenames) > 1:
             raise Exception('Image stacks not supported.')
-        data_file_path = filenames[0]
+        data_path = filenames[0]
         # fileName is relative to the state file location, so convert to
         # absolute path.
-        data_file_path = os.path.abspath(
-            os.path.join(os.path.dirname(state_file_path), data_file_path))
-        if not data_file_path.lower().endswith('.emd'):
+        data_path = os.path.abspath(
+            os.path.join(os.path.dirname(state_file_path), data_path))
+        if not data_path.lower().endswith('.emd'):
```

Comment:
So I noticed that when it is reading from the state file, it requires the file to be EMD. But below ([this block](https://github.com/OpenChemistry/tomviz/pull/1928/files#diff-f4a51f1c4f8109ed11a0e42e6aac1c32R96)), there doesn't seem to be an EMD requirement. Is that intentional?

Reply:
Well, in the case of a directory being provided we glob for *.emd, in the other case we would try to open the file as an hdf5 file which, will probably raise an exception. i can add in another check for the file ending if you think this is useful.

Reply:
No, I was just wondering. It's up to you. Looks good otherwise.

Reply:
I moved the check further down, so now it should catch both cases.